internal/resources/grafana/resource_alerting_rule_group.go

```diff
@@ -129,9 +129,11 @@ This resource requires Grafana 9.1.0 or later.
 										Description: "An optional identifier for the type of query being executed.",
 									},
 									"model": {
-										Required:    true,
-										Type:        schema.TypeString,
-										Description: "Custom JSON data to send to the specified datasource when querying.",
+										Required:     true,
+										Type:         schema.TypeString,
+										Description:  "Custom JSON data to send to the specified datasource when querying.",
+										ValidateFunc: validateModelJSON,
```

Comment:
```suggestion
										ValidateFunc: validation.StringIsJSON,
```

You can use the standard validation lib for that 

`"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"`